app/modules/code_provider/provider_factory.py

```
@@ -0,0 +1,454 @@
+        # GitHub repos must be in "owner/repo" format
+        if provider_type == "github" and "/" not in repo_name:
+            logger.error(
+                f"Invalid GitHub repository name: '{repo_name}'. "
+                f"GitHub repos must be in 'owner/repo' format. "
+                f"If this is a local repository, ensure repo_path is set in the database."
+            )
+            raise ValueError(
+                f"Invalid repository name '{repo_name}'. "
+                f"GitHub repos must be in 'owner/repo' format (e.g., 'facebook/react'). "
+                f"For local repositories, use the full filesystem path."
+            )
```

Ruff (0.14.3)

Avoid specifying long messages outside the exception class

(TRY003)